omp/matrix/coo_kernels.cpp

```diff
@@ -103,15 +104,45 @@ void spmv2(std::shared_ptr<const OmpExecutor> exec,
            const matrix::Coo<ValueType, IndexType> *a,
            const matrix::Dense<ValueType> *b, matrix::Dense<ValueType> *c)
 {
-    auto coo_val = a->get_const_values();
-    auto coo_col = a->get_const_col_idxs();
-    auto coo_row = a->get_const_row_idxs();
-    auto num_cols = b->get_size()[1];
-
-#pragma omp parallel for
-    for (size_type j = 0; j < num_cols; j++) {
-        for (size_type i = 0; i < a->get_num_stored_elements(); i++) {
-            c->at(coo_row[i], j) += coo_val[i] * b->at(coo_col[i], j);
+    const auto coo_val = a->get_const_values();
+    const auto coo_col = a->get_const_col_idxs();
+    const auto coo_row = a->get_const_row_idxs();
+    const auto num_rhs = b->get_size()[1];
+    const auto sentinel_row = a->get_size()[0] + 1;
+    const auto nnz = a->get_num_stored_elements();
+    const auto num_threads = omp_get_max_threads();
+    const auto work_per_thread = ceildiv(nnz, num_threads);
+
+#pragma omp parallel num_threads(num_threads)
```

Comment:
I didn't bother to dive too deep into the OpenMP Specs to see if num_threads is necessary here. Probably not (:

Reply:
True, you don't need it.

Reply:
I think still, the most stable solution would be to compute the work per thread inside the parallel region, then we know how many threads were actually spawned. Otherwise we might run into issues if this gets called from within a parallel region?